extensions/wikia/AdEngine/js/slot/adUnitBuilder.js

```diff
@@ -18,7 +19,61 @@ define('ext.wikia.adEngine.slot.adUnitBuilder', [
 		].join('/');
 	}
 
+	function findSlotGroup(map, slotName) {
+		var result = Object.keys(map).filter(function (name) {
+			return map[name].indexOf(slotName) !== -1;
+		});
+
+		return result.length === 1 ? result[0] : null;
+	}
+
+	function getGroup(slotName) {
+		var map = {
+			'LB': ['TOP_LEADERBOARD', 'MOBILE_TOP_LEADERBOARD'],
+			'MR': ['TOP_RIGHT_BOXAD'],
+			'SKY': ['LEFT_SKYSCRAPER_2', 'LEFT_SKYSCRAPER_3'],
+			'PF': ['PREFOOTER_LEFT_BOXAD', 'PREFOOTER_MIDDLE_BOXAD', 'PREFOOTER_RIGHT_BOXAD', 'MOBILE_PREFOOTER'],
+			'PX': ['INVISIBLE_SKIN', 'INVISIBLE_HIGH_IMPACT_2'],
+			'HiVi': ['INCONTENT_BOXAD_1', 'MOBILE_IN_CONTENT']
+		}, result;
+
+		result = findSlotGroup(map, slotName);
+		// OTHER: 'BOTTOM_LEADERBOARD', 'MOBILE_BOTTOM_LEADERBOARD', 'INCONTENT_LEADERBOARD', 'EXIT_STITIAL_BOXAD_1'
+		return result ? result : 'OTHER';
+	}
+
+	function getDevice(params) {
+		var result = 'unknown';
+
+		if (params.skin === 'oasis') {
+			result = browserDetect.isMobile() ? 'tablet' : 'desktop';
+		} else if (params.skin === 'mercury' || params.skin === 'mobile-wiki') {
+			result = 'smartphone';
+		}
+
+		return result;
+	}
+
+	function buildNew(src, slotName, passback) {
```

Comment:
What do you think about moving this to new module and keep exactly the same API for both adUnitBuilders? Then we can pass new module in:
https://github.com/Wikia/app/pull/12745/files#diff-31d08cf2d202c80ee219ba9cc3e46e03R43
And keep 'decision' logic in one place: 
https://github.com/Wikia/app/pull/12745/files#diff-4addc132482932ea704402dd5c7d7746R67